### PR TITLE
Add Genio provider to contrib area (Infra)

### PR DESCRIPTION
**Pull request by pieqq**
## Description

The Genio provider has been imported and placed in the contrib area. I cherry-picked the commits made by @baconYao in his branch, and made a few other changes:

- moved the provider to `contrib/`
- modified its namespace to `com.canonical.contrib`
- modified some jobs to import the `executable` job from the resource provider before using it (since Genio and resource providers now live in two separate namespaces)

## Resolved issues

https://warthogs.atlassian.net/browse/RTW-251

## Documentation

N/A

## Tests

I ran `./manage.py validate` to check the provider was valid.

### CODE REVIEW

**Comment by baconYao** [comment]
Hi @pieqq, thanks your efforts and there's only one question is that do we still need the `py3.5`, `py3.6` and `py3.8` workflow checking even Genio project starts from `classic 22.04` and  `UC22`?

**Review by kissiel — APPROVED** [review]
Let's land!

File: .github/workflows/tox-contrib-provider-ce-oem.yaml [hunk 1]
```diff
@@ -1,18 +1,11 @@
...
 
 on:
-  push:
-    branches: [ main ]
-    paths:
-      - contrib/checkbox-provider-ce-oem/**
-  pull_request:
-    branches: [ main ]
-    paths:
-      - contrib/checkbox-provider-ce-oem/**
```
Comment (LiaoU3) [inline .github/workflows/tox-contrib-provider-ce-oem.yaml:11]: @pieqq Do you know why you remove these?

**Review by LiaoU3 — COMMENTED** [review]
@pieqq Please see my question here. Thank you!

**Pull request merged**